### PR TITLE
fix(core): onDrop and onPaste only registered when defined #5681

### PR DESCRIPTION
**Pull request by nperez0111**
## Changes Overview

This updates the `onDrop` and `onPaste` event callbacks to only ever be registered if they exist on the editor instance at creation.
It also refactors them to be extensions rather than separate Prosemirror plugins which would causes plugins to be destroyed and recreated immediately
## Checklist
- [ ] I have created a [changeset](https://github.com/changesets/changesets) for this PR if necessary.
- [ ] My changes do not break the library.
- [ ] I have added tests where applicable.
- [ ] I have followed the project guidelines.
- [ ] I have fixed any lint issues.

### CODE REVIEW

File: packages/core/src/Editor.ts [hunk 1]
```diff
@@ -112,14 +111,8 @@ export class Editor extends EventEmitter<EditorEvents> {
...
-
-    if (this.options.onPaste) {
-      this.registerPlugin(PastePlugin(this.options.onPaste))
-    }
-
-    if (this.options.onDrop) {
-      this.registerPlugin(DropPlugin(this.options.onDrop))
-    }
+    this.on('drop', ({ event, slice, moved }) => this.options.onDrop(event, slice, moved))
+    this.on('paste', ({ event, slice }) => this.options.onPaste(event, slice))
```
Comment (nperez0111) [inline packages/core/src/Editor.ts:115]: I figured, that while I was here, I may as well promote these to be actual editor events that follow the same pattern as others. This allows flexibility to register multiple handlers to listen to the same event

**Review by bdbch — APPROVED** [review]
LGTM

File: packages/core/src/Editor.ts (hunk 1 above)
Comment (svenadlung) [reply packages/core/src/Editor.ts:115]: @nperez0111 This resolves [this](https://github.com/ueberdosis/tiptap/discussions/5613), right?

Comment (nperez0111) [reply packages/core/src/Editor.ts:115]: @svenadlung it does 😉

**Pull request merged**